### ISSUE TITLE
Add Fade when changing states in the State effect

### ISSUE THREAD

**Issue by SopherConklin** [xLightsSequencer/xLights#4764]
So, I was making some states with the Spiders in the ShowStopper Spinner.   I wanted to color the various parts of the spiders and show them moving a bit, so the States effect seemed a natural.

I was testing using a 1 sec metronome to move the spider every second.  The problem is that there are many common pixels between the adjacent spiders, so when you just go through the states, it seems like a jumbled mess, and doesn't really resemble a spider moving around.

 I found that if I made a new States effect every second, then I could give each of them a 0.25 sec fade out and then they looked much much better - they started looking like a spider moving around.   Now, any song that I want to use this I need to paste in 1 State effect every second, which isn't very efficient.  And lots of State effects are faster than 1 second, so that could become a LOT of State effects.

So, I'm trying to be a responsible citizen by identifying useful possible features.  I know you guys already have full plates, but if I don't record this, I'll forget about it a day from now, so at least it's on the list.

Video showing an application for this:
https://youtu.be/m_6lHj1D8SY

Sequence from the video:
https://www.dropbox.com/scl/fi/j1i9a1x1entt5q04ejqcf/Spider-states.xsqz?rlkey=hkrmyepmri1k4jat6130i1dgr&st=bas5jrfc&dl=0

P.S.  I almost forget - if you have a long list of nodes in a line in the States creation screen, and you check Force Custom Colors, then the node numbers overwrite the Color box until you double click on it and enter a color.   This is minor, but it really confused me and I had to go to the Zoom Room to make sure I hadn't screwed something up and learned I could just double click the node numbers in the Color box and it would work fine.   It is confusing and distressing, so consider keeping the Nodes in the Nodes box and not letting them overwrite the Color box, even if it's empty.

Crap, I never said what I came to say - instead of inserting 1 State effect every second to get a fade at each change of state, it would be much more useful and versatile if we incorporated an optional user-specified fade every time the state changes within the State effect itself.  In this case, I could add 1 State effect that spanned the entire time and every time the state changed (spider moved) there would be a fade out.   Since we don't know a priori how long each state will be, maybe we could specify the fade as a percentage of the state time.  Lots of things look better with a fade.

**Comment by SopherConklin** [comment]
P.S.  I almost forget - if you have a long list of nodes in a line in the States creation screen, and you check Force Custom Colors, then the node numbers overwrite the Color box until you double click on it and enter a color.   This is minor, but it really confused me and I had to go to the Zoom Room to make sure I hadn't screwed something up and learned I could just double click the node numbers in the Color box and it would work fine.   It is confusing and distressing, so consider keeping the Nodes in the Nodes box and not letting them overwrite the Color box.

**Comment by SopherConklin** [comment]
Crap, I never said what I came to say - instead of inserting 1 State effect every second to get a fade at each change of state, it would be much more useful and versatile if we incorporated an optional user-specified fade every time the state changes within the State effect itself.  In this case, I could add 1 State effect that spanned the entire time and every time the state changed (spider moved) there would be a fade out.   Since we don't know a priori how long each state will be, maybe we could specify the fade as a percentage of the state time.  Lots of things look better with a fade.
